code/datums/craft/gun_parts.dm

```diff
@@ -20,6 +20,15 @@
 	var/mechanism_attached = FALSE
 	var/barrel = /obj/item/part/gun/barrel
 	var/barrel_attached = FALSE
+	var/serial_type = ""
+
+
+/obj/item/part/gun/frame/New(loc, ...)
+	. = ..()
+	var/obj/item/gun/G = new result(null)
+	if(G.serial_type)
+		serial_type = G.serial_type
```

Comment:
You can use `initial()` here instead of instancing a new gun (or at least delete the new gun afterwards)

Reply:
That gun instance are being deleted when proc is finished (hence null spawn location).
Not sure if initial could be called on a type, but I'll try.

Reply:
it can be called on any var

Reply:
Also gun's proper serials only generated on new() call.

Reply:
> But `var/obj/item/gun/G` is a variable only in the scope of this proc

It is, but I don't understand where you're going with this.

> it can be called on any var

Doesn't really work for me.

Reply:
It won'd _qdel_ it exactly, but since there is no references, byond garbage collector won't go scan everything for references.